thirdparty/deto_async/async/internal/queuedinvoker.cpp

```diff
@@ -22,8 +22,12 @@ void QueuedInvoker::invoke(const std::thread::id& th, const Functor& f, bool isA
 
 void QueuedInvoker::processEvents()
 {
-    std::lock_guard<std::recursive_mutex> lock(m_mutex);
-    Queue& q = m_queues[std::this_thread::get_id()];
+    Queue q;
+    {
+        std::lock_guard<std::recursive_mutex> lock(m_mutex);
+        q = m_queues[std::this_thread::get_id()];
```

Comment:
might make sense to use extract here

Reply:
but I'm curious what was the problem?

Reply:
Unfortunately, my stdlib implementation no has method `value` for `node_type`
```
auto n = m_map.extract(key);
T val = n.value(); <-  not compiled

output:
/home/igor/Dev/MuseScore/thirdparty/deto_async/async/internal/queuedinvoker.cpp:30:15: error: ‘class std::_Node_handle<std::thread::id, std::pair<const std::thread::id, std::queue<std::function<void()> > >, std::allocator<std::_Rb_tree_node<std::pair<const std::thread::id, std::queue<std::function<void()> > > > > >’ has no member named ‘value’
         q = n.value();
```

Ubuntu 18.04 gcc 7.5 (same as on CI)

Reply:
> but I'm curious what was the problem?

I found, that when the application starts, after a few seconds, a freeze appears for about 5-10 seconds.
Freeze calls because of attempts to lock this mutex when call "invoke", that is, someone, another thread, holds this mutex in this place when processing messages.
that is, there is a very long handler that takes a long time to execute and, accordingly, holds the mutex.
In fact, we need to hold the mutex while we process the messages. We only need a mutex to pop the message queue.

Reply:
it's confusing, but there is no method "value()" for a node_handle of std::map, it's called "mapped()". It's a value from <key, value> pair in node_handle from std::map

Reply:
thanks, added use `extract`